tests/simulator/test_advance.py

```
@@ -287,21 +287,113 @@ def base_test_overlaped_fields_are_equal(self, datahier, coarsest_time):
+
+                        if box.ndim == 2:
+                            failed_i, failed_j = np.where(
+                                np.abs(slice1 - slice2) > 5.5e-15
+                            )
+
+                            def makerec(
+                                lower, upper, dl, fc="none", ec="g", lw=1, ls="-"
+                            ):
+                                origin = (lower[0] * dl[0], lower[1] * dl[1])
+                                sizex, sizey = [
+                                    (u - l) * d for u, l, d in zip(upper, lower, dl)
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)